bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/LcnModuleHandler.java

```diff
@@ -290,6 +291,22 @@ public void updateChannel(LcnChannelGroup channelGroup, String channelId, State
         if (converter != null) {
             convertedState = converter.onStateUpdateFromHandler(state);
         }
+
+        if (channelGroup.equals(LcnChannelGroup.BINARYSENSOR)) {
+            Channel channel = thing.getChannel(channelUid);
+            if (channel != null) {
+                Object invertConfig = channel.getConfiguration().get("invertOpenClosed");
+                boolean invertOpenClosed = invertConfig instanceof Boolean && (boolean) invertConfig;
+                if (invertOpenClosed) {
+                    if (state.equals(OpenClosedType.OPEN)) {
+                        convertedState = OpenClosedType.CLOSED;
+                    } else {
+                        convertedState = OpenClosedType.OPEN;
```

Comment:
Can you move this into `LcnModuleBinarySensorSubHandler` similar to the inversion of the roller shutter up/down in `LcnModuleRollershutterRelaySubHandler`?

Reply:
If one delegates `updateChannel` to `LcnModuleBinarySensorSubHandler` (by also adding a corresponding method to `AbstractLcnModuleSubHandler`) one needs to create `ChannelUID` for calling `updateState` there, but method `LcnModuleHandler.createChannelUid` is private. Because `LcnModuleHandler` and `LcnModuleBinarySensorSubHandler` don't have a common superclass: should we move `LcnModuleHandler.createChannelUid` to a helper class?

Reply:
`BaseThingHandler.updateState` is not visible from `AbstractLcnModuleSubHandler`. So one can not delegate channel update handling to a sub handler similar to the rollershutter command handling (without revealing `BaseThingHandler.updateState` in `LcnModuleHandler`). We could delegate state conversion within `LcnModuleHandler.updateChannel` to a converter (similar to `Converter.onStateUpdateFromHandler(State)`). This would require `Converter` to be an (abstract) superclass or better interface for arbitrary converters which are initialized in `LcnModuleHandler.initialize`. At the moment `Converter` is specialized to deal with variables only.

What do you think?

Reply:
If I see correctly, `BaseThingHandler.updateState()` is invoked via `LcnModuleHandler.updateChannel()` -> `AbstractLcnModuleSubHandler.fireUpdate()` -> `LcnModuleBinarySensorSubHandler.handleStatusMessage()`. I think the conversion could happen in LcnModuleBinarySensorSubHandler:55.

Reply:
Thanks for your feedback! If I didn't miss something when following your suggestion channel update (and status message handling resp.) would differ between binary sensors and variables. So from a design perspective I think we should choose between using the converter approach or uniquely handle value conversion within the sub-handler's `handleStatusMessage`.

Would you like to decide how to uniquely handle value conversion and adapt the code correspondingly? Perhaps it would make sense to handle this on a separate enhancement/pull request due to the necessary refactorings.

Reply:
Although I'm a fan a of abstracting things, I wouldn't combine conversion of raw data into human readable values and the inversion of binary sensor states. First is a conversion, the other is a manipulation. From the functional perspective, I think they have to less in common to merge these functions.

Reply:
WDYT?

Reply:
From my point of view there is no conceptual difference between converting raw variable data (i.e. lcn value) into temperature, speed etc. and converting raw lcn binary sensor data (0 or 1) into open/closed states. In my opinion in both cases conversion applies semantic to raw sensor data leading to human readable values.

Reply:
Ok, you convinced me. I refactored the conversion code and made `Converter` an interface. Do you want to implement an inversion Converter?

Can you give me access rights to merge the changes into your branch? Alternatively, you could merge my `lcn-converter` branch into yours. Before doing so you have to rebase this PR's branch to bb6722c.

Reply:
I added an `InversionConverter`. For this I moved `Converter.onCommandFromItem(double)` and `Converter.onCommandFromItem(QuantityType<?>)` down the hierarchy to `ValueConverter` because I think these methods make only sense when converting numerical values. The `InversionConverter` might later additionally be used e.g. for visualizing channels with inversed `OnOff` or `UpDown` states.